test/topology_experimental_raft/test_tombstone_gc.py

```diff
@@ -44,3 +50,84 @@ async def test_default_tombstone_gc_does_not_override(manager: ManagerClient, rf
         async with new_test_table(cql, keyspace, "p int primary key, x int", " with tombstone_gc = {'mode': 'disabled'}") as table:
             await cql.run_async(f"ALTER TABLE {table} add y int")
             check_tombstone_gc_mode(cql, table, "disabled")
+
+
+@pytest.mark.asyncio
+@pytest.mark.xfail(reason="issue #15607")
+async def test_group0_tombstone_gc(manager: ManagerClient):
+    """
+    Regression test for #15607.
+
+    Test that the tombstones are being cleaned after the group 0 catching up.
+
+    Arrange:
+        - start 3 nodes
+    Act:
+        - create new group0 tombstones by updating the schema (create/alter/delete randome tables)
+    Assert:
+        - the tombstones are cleaned up eventually
+    """
+    cmdline = [
+        '--enable-cache', '0',
+    ]
+    servers = [await manager.server_add(cmdline=cmdline) for _ in range(3)]
+
+    cql = manager.get_cql()
+    hosts = [(await wait_for_cql_and_get_hosts(cql, [s], time.time() + 60))[0]
+             for s in servers]
+
+    # create/alter/drop a few tables
+    async def alter_system_schema(table_count=3):
+        for _ in range(table_count):
+            async with new_test_table(cql, keyspace, "p int primary key, x int") as table:
+                await cql.run_async(f"ALTER TABLE {table} add y int")
+
+    def get_tombstone(row):
+        if row.metadata is None:
+            return None
+        metadata = json.loads(row.metadata)
+        return metadata.get("tombstone")
+
+    async def list_tombstones(tombstone_mark, host):
+        tombstones = {}
+        for tbl in ("tables", "columns"):
+            res = list(cql.execute(f"SELECT * FROM MUTATION_FRAGMENTS(system_schema.{tbl})", host=host))
+            tombstones[tbl] = []
+            for row in res:
+                tombstone = get_tombstone(row)
+                if tombstone and datetime.fromtimestamp(float(tombstone["timestamp"])/1_000_000, timezone.utc) < tombstone_mark:
+                    tombstones[tbl].append(tombstone)
+        return tombstones
+
+    async def tombstone_gc_completed(tombstone_mark):
+        # keep generating changes to generate new state_ids
+        # (needed for the gossip protocol to keep updating the state)
+        await alter_system_schema(table_count=1)
```

Comment:
> We shouldn't have to keep updating the state_id. The state_id that comes with the last schema change, even if it created a bunch of tombstones, should be sufficient. Because that state_id stores a timestamp that is greater or equal to to the timestamps of all tombstones that it (and previous schema changes) created.

After more thought that could actually be false since gc_clock has worse resolution, I think you mentioned it. (The gc time we calculate gets truncated down) but in that case only a single additional schema change should suffice. But however many schema changes the test does, it should definitely make sure that it operates on a completely disjoint set of rows -- unlike `alter_system_schema` which underneath will reuse ks/table names.

Reply:
Yeah, it is needed to do any change at least once after the last tombstone expired so that a new state_id is propagated and allows to remove the tombstones. However you are right that the change should be only necessary once, so I changed the test accordingly, also making sure that the extra update is using a different keyspace/table (as it looks indeed that the tombstones would just be overwritten otherwise).